### PR TITLE
Handle static libraries in pods

### LINKED ISSUE #299: Static libraries in resolved cocoapod frameworks are ignored

### PR DESCRIPTION
**Pull request by britzl**
Fixes #299

### CODE REVIEW

File: server/src/main/java/com/defold/extender/Extender.java [hunk 1]
```diff
@@ -381,20 +381,25 @@ private List<String> getFrameworks(ResolvedPods resolvedPods) {
         return frameworks;
     }
 
-    private List<String> getFrameworkPaths(File dir) {
-        List<String> frameworkPaths = new ArrayList<>();
+    // Get a list of subfolders matching the current platform
+    private List<String> getPlatformPaths(File dir) {
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/Extender.java:385]: Helper function to get the matching platform paths for a folder.

File: server/src/main/java/com/defold/extender/Extender.java [hunk 2]
```diff
@@ -404,6 +409,23 @@ private List<String> getFrameworkPaths(ResolvedPods resolvedPods) {
         return frameworkPaths;
     }
 
+    private List<String> getFrameworkStaticLibs(ResolvedPods resolvedPods) {
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/Extender.java:412]: Get the static libraries (*.a) for all frameworks

File: server/src/main/java/com/defold/extender/Extender.java [hunk 3]
```diff
@@ -470,6 +492,18 @@ private List<String> getExtLocalIncludeDirs(File dir) {
         return includes;
     }
 
+    private List<String> getFrameworkStaticLibIncludeDirs(ResolvedPods pods) {
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/Extender.java:495]: Get all headers for static libraries included in pod frameworks

File: server/src/main/java/com/defold/extender/ExtenderUtil.java [hunk 4]
```diff
@@ -582,6 +582,28 @@ static public List<String> collectFilesByName(File dir, String re) {
         return result;
     }
 
+    // Find all static libs (*.a) in a dir
+    // Returns list of lib names without initial "lib" and ".a" extension
+    static public List<String> collectStaticLibsByName(File dir) {
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/ExtenderUtil.java:587]: Helper function to collect static lib names

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 5]
```diff
@@ -208,29 +234,27 @@ public class PodSpec {
...
+            sb.append("  linkflags: " + linkflags + "\n");
+            sb.append("  weak_frameworks: " + weak_frameworks + "\n");
+            sb.append("  resources: " + resources + "\n");
+            sb.append("  frameworks: " + frameworks + "\n");
+            sb.append("  vendoredframeworks: " + vendoredframeworks + "\n");
+            sb.append("  libraries: " + libraries + "\n");
+            sb.append("  parentSpec: " + ((parentSpec != null) ? parentSpec.name : "null") + "\n");
             for (PodSpec sub : subspecs) {
-                sb.append(sub.toString(indentation + "  "));
+                sb.append("  subspec: " + sub.name + "\n");
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:254]: It becomes extremely messy when also printing the full subspecs. Better to only print the subspec names.

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 6]
```diff
@@ -739,13 +804,19 @@ private PodSpec createPodSpec(JSONObject specJson, PodSpec parent, File podsDir,
 
         // parse subspecs
         // https://guides.cocoapods.org/syntax/podspec.html#subspec
+        JSONArray defaultSubspecs = getAsJSONArray(specJson, "default_subspecs");
```
Comment (britzl) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:807]: Make sure to respect the default_subspec option and only use the subspec(s) specified in this option.

File: server/src/main/java/com/defold/extender/Extender.java [hunk 7]
```diff
@@ -991,8 +1026,13 @@ private void getProjectPaths(Map<String, Object> mainContext, Map<String, Object
...
         extLibs.addAll(ExtenderUtil.collectFilesByName(buildDirectory, platformConfig.stlibRe));
-        extFrameworks.addAll(getFrameworks(resolvedPods));
-        extFrameworkPaths.addAll(getFrameworkPaths(resolvedPods));
+
+        if (resolvedPods != null) {
+            extFrameworks.addAll(getFrameworks(resolvedPods));
+            extFrameworkPaths.addAll(getFrameworkPaths(resolvedPods));
+            extLibs.addAll(getFrameworkStaticLibs(resolvedPods));
+            extLibPaths.addAll(getFrameworkStaticLibPaths(resolvedPods));
+        }
```
Comment (AGulev) [inline server/src/main/java/com/defold/extender/Extender.java:1035]: it's getting bigger and bigger )

**Review by AGulev — APPROVED** [review]
Wow! This is huge. 
But it looks good,

**Pull request merged**